src/layout/shell/UserProfile.tsx

```diff
@@ -43,7 +40,11 @@ export function UserProfile() {
   const id = useId();
 
   const { openModal } = useModal();
-  const { name, email, isDummy } = useUserProfile();
+  const userId = useUserProfile((state) => state.id);
+  const name = useUserProfile((state) => state.name);
+  const email = useUserProfile((state) => state.email);
```

Comment:
Why not this, is it supposed to trigger re-renders more often?  I believe the user object shouldn't change at all, or is there anything else triggering state changes?
`const { id: userId, name, email } = useUserProfile((state) => state);`